src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -69,6 +69,26 @@
       (log/info
        (i18n/trs "Writing jruby profiling output to ''{0}''" real-profiler-output-file)))))
 
+(schema/defn ^:always-validate set-ruby-encoding :- jruby-schemas/ConfigurableJRuby
+  "Sets the K code, source encoding, and external encoding of the JRuby instance to
+  the supplied encoding."
+  [kcode :- KCode
+   jruby :- jruby-schemas/ConfigurableJRuby]
+  (let [encoding-string (str (.getEncoding kcode))]
+    ;; RubyInstanceConfig exposes the encoding setters directly, while the
+    ;; ScriptingContainer does not.
+    (if (instance? RubyInstanceConfig jruby)
```

Comment:
Do you think it is worth being explicit here and using `cond` instead of `if` so that we can test for both `RubyInstanceConfig` and `ScriptingContainer`?